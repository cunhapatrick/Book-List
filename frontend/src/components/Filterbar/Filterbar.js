import React from 'react';
import PropTypes from 'prop-types';

// CSS
import  {
  Container,
  InputData,
} from './Filterbar.style';

const Filterbar = ({
  iData,
  eData,
  handleIData,
  handleEData,
  total,
}) => (
  <Container>
    <span>Filtrar ano de publicação:</span>
    <div>
      <InputData
        value={iData}
        onChange={handleIData}
        type="text"/>
      <i className="far fa-calendar-alt"/>
    <span>até</span>
      <InputData
        value={eData}
        onChange={handleEData}
        type="text"/>
      <i className="far fa-calendar-alt"/>
    </div>
    <span>
      {total} resultados encontrados
    </span>
  </Container>
);

Filterbar.propTypes = {
  iData: PropTypes.number.isRequired,
  eData: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired,
  handleIData: PropTypes.func.isRequired,
  handleEData: PropTypes.func.isRequired,
}

export default Filterbar;