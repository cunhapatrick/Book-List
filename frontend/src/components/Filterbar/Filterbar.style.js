import styled from 'styled-components';

export const Container = styled.div`
  width: 100%;
  height: 45px;
  padding: 10px 20px;
  border-bottom: 1px solid black;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex-direction: row;

  span {
    flex: 0 1 10%;
    font-size: 16px;
    margin-right: 10px;
  }

  div {
    flex: 2;
    i, span {
      margin-right: 10px;
    }

    i {
      font-size: 22px;
    }
  }
`;

export const InputData = styled.input`
  width: 50px;
  height: 30px;
  margin-right: 10px;
`;