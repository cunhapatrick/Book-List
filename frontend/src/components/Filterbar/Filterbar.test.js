import React from 'react';
import { shallow } from 'enzyme';
import Filterbar from './Filterbar';

describe('Testing Filterbar Component', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <Filterbar
        iData={1998}
        eData={2019}
        total={10}
        handleIData={() => {}}
        handleEData={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});