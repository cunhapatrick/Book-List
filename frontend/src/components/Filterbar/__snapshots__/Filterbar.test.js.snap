// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Filterbar Component should render properly 1`] = `ShallowWrapper {}`;
